contrib/nginx/README.md

```diff
@@ -1,12 +1,56 @@
-# Nginx
+# Serving everything from one Nginx server
 
-This provides a experiment of how to set up a discovery server by nginx. It is useful for integration test.
+Publishers who intend to serve discoverable images via the protocols in this repository, but who only want to serve static content can use [Nginx][].
+This example walks you through one possible approach.
 
-Developer can build up via Dockerfile and run a discovery server:
+The [Nginx configuration file](nginx.conf) is fairly straightforward, redirecting HTTP requests to HTTPS, and serving HTTPS out of `/srv/example.com` with some media type overrides to get proper typing for [ref-engines objects](../../xdg-ref-engine-discovery.md#ref-engines-objects) and [OCI indexes][index].
+
+The [ref-engines object](../../xdg-ref-engine-discovery.md#ref-engines-objects) for the [OCI Well-Known URI Ref-Engine Discovery specification](../../well-known-uri-ref-engine-discovery.md) is served from `/srv/example.com/.well-known/oci-host-ref-engines` with example content like [this](example.com/.well-known/oci-host-ref-engines).
+
+With the [example ref-engines object](example.com/.well-known/oci-host-ref-engines), consumers will attempt to resolve image names matching the `example.com/app#…` family of [host-based image names](../../host-based-image-names.md) via an [OCI Index Template](../../index-template.md) ref engine at `https://example.com/oci-index/app`.
+That [index][] is served from `/srv/example.com/oci-index/app` with example content like [this](example.com/oci-index/app).
+
+The `org.opencontainers.image.ref.name` value in [the example index](example.com/oci-index/app) assumes consumers will only be attempting to match the `fragment` and not the full image name; image-spec does not currently provide guidance on this point.
+
+The `casEngines` entry in [the example index](example.com/oci-index/app) suggests consumers retrieve CAS blobs from `https://example.com/oci-cas/{algorithm}/{encoded:2}/{encoded}`.
+Supply those blobs at `/srv/example.com/oci-cas/{alg}/{prefix}/{encoded:2}/{encoded}`.
```

Comment:
Oops, this should have been `{algorithm}/{encoded:2}/{encoded}`, not `{alg}/{prefix}/{encoded:2}/{encoded}`.

Reply:
Filed a fix in #52.